### ISSUE TITLE
Docs4Doge : Document dogecoin integration with third party into applications or website

### ISSUE THREAD

**Issue by AbcSxyZ** [dogecoin/dogecoin#2582]
![Capture d’écran de 2021-09-22 19-39-56](https://user-images.githubusercontent.com/34010605/134394588-3fada82d-8b7e-4bef-979b-49a7b55bfb34.png)

To facilitate Dogecoin integration, it would be great to give documentation about ways to enable Dogecoin payment for an application.

There are multiple ways to achieve this. By managing the node yourself for sure, but you can rely on different services too. Multiple service providers build tools on top of this, to facilitate integration, even for multiple cryptocurrency integration.

Are you agree to add some documentation to redirect to services like [Bitpay API](https://bitpay.com/api/#rest-api), [Coinbase Commerce API](https://commerce.coinbase.com/docs/api/), [Flexa](https://developer.flexa.network/) or whatever can be recommended ?

Having, for example, something structured like that :
> ## Integrate dogecoin payment
> #### Self hosting
> .....
> #### Service provider
> ....

And if you know any other way to have payment for a developer ?

*Related to discussion https://github.com/dogecoin/dogecoin/discussions/2430 about community repositories, how to include actors of the ecosystem and [Docs4Doge initiative](https://github.com/dogecoin/dogecoin/issues/2271).*

**Comment by patricklodder** [comment]
> Are you agree to add some documentation to redirect to services like Bitpay API, Coinbase Commerce API, Flexa or whatever can be recommended ?

No. That has nothing to do with Dogecoin Core. You could work on something like that for, say, https://github.com/dogecoin/dogecoin.com though.

**Comment by AbcSxyZ** [comment]
Its services built on top of it to facilitate is usage, I see it as definitively related to Dogecoin core. It's how people will be able to integrate dogecoin to their website, through their own node, or through a service built on top of Dogecoin Core.

I guess a lot of devs don't/won't host their own node to integrate Dogecoin, but (will) rely on a third party. Using this kind of tools would be the way for them to integrate Dogecoin.

Someone who is asking how to integrate dogecoin payment, if he come here, will only have the solution of hosting a node, which is not (I suppose) how a lot of business do this.

Early or later, transaction happen on the blockchain, through a json-RPC or Coinbase Commerce.

**Comment by AbcSxyZ** [comment]
Like it, they can be aware of this solution. Devs may use service providers for an easy way to use the crypto, while more important project, with more resources will see also this solution instead of just viewing an API from any exchange or payment application.

Trying to have people coming here first, watching different solution from node to services, and then do the choice. Instead of directly use the API from Coinbase and use it, they can be more open-minded and informed about what is possible.

**Comment by patricklodder** [comment]
Note that I'm not against your goal, on the contrary: I like it. I just do not think that this is the right place to host it.

> Its services built on top of it to facilitate usage, I see it as definitively related to Dogecoin core.

Not all services are built on top of Dogecoin Core. Over the years, I personally have built Dogecoin services using p2p libraries like [bitcore](https://github.com/bitpay/bitcore), [btcd](https://github.com/btcsuite/btcd) and [toshi](https://github.com/martindale/toshi) that do not have any dogecoind running anywhere. And there are SPV client libraries like [libdohj](https://github.com/dogecoin/libdohj) and [dogecoin-spv-node](https://github.com/BitcoinAmiens/dogecoin-spv-node) that can fully operate without a Dogecoin Core node. So not everything is about Dogecoin Core.

The only way these software suites are related to Dogecoin Core is that their development and/or integration may require a deep look at the code here, to see how particular protocols, data structures and mechanisms are working. This doesn't mean that the entire ecosystem of Dogecoin implementations and services has to be managed from this repository; in fact, I think that's counter-productive.

> Trying to have people coming here first [..]

This puts development of this software on the critical path for everything, unnecessarily, creating friction in places where there shouldn't be. Any feature of this repository also must follow the release and review processes and this burdens resources that do not have to be involved, and slows down the development process of the information you're proposing to maintain and expose here. Besides, Dogecoin is permissionless, but this repository is permissioned. So this is why I think your proposal is not a good fit, at all.

If you can do something without involving this software repository, you should. It is merely the reference implementation and the main reason (too) many people come here with things that do not belong here is because the rest of the ecosystem is still poor in diversity and quality. If you want to make Dogecoin better, you can do it without anyone's permission, off this repository, and make the entire Dogecoin ecosystem more diverse.

Because what you're proposing, instead of solving the problem of too much centralization, is making this repository a one-stop shop for everything related to Dogecoin, and we would further centralize, give more power to people that already have way too much of that (including yours truly) and do a disservice to the long-term success of Dogecoin. Because let's be real: Dogecoin is not a business, it's a public domain phenomenon, made so by the original project over 7 years ago. If you want it to be "the people's coin", then the people must not delegate all control to a select few that just happened to have inherited some rights on a GitHub repo. The potential for control should be minimized, not maximized.

I understand that tapping into the traffic this repository currently attracts seems to make sense from a perspective of maximizing reach (i.e. trying to growth hack), but the website reaches more people and is technically still a community effort. It will at least solve the issue of being release-bound by using that instead, so that's why I suggested that. Even that is not ideal though because it's still centralized. We need diversity.

Please, consider utilizing independent resources to reach your goals. If you need help in any way to get started, be it advice, review time, some help to fund hosting, anything basically, feel free to reach out. I'm willing to help, but not here.

**Comment by AbcSxyZ** [comment]
I think the difficulty is that the Dogecoin ecosystem is a vast variety of projects, and there is a difficulty to define the right limits for the type of project we could link to for developers of and around Dogecoin Core. 

You brought this discussion https://github.com/dogecoin/dogecoin/discussions/2430 about "community repositories", you were speaking about listing SDKs for the RPC API. I share this idea that it is a good thing to speak about inside the documentation, it's directly useful to develop application with your node. Personally, I would prefer to use a python sdk instead of the cli to interact with it, having this info while installing my node and seeing how it works is great.

In my mind, when I'm speaking about some service providers who will give "direct api for the core", payment providers, **it's still for technical users who would like to implement functionality of the core, to send and view transactions.** It's not about including every project related to Dogecoin.

People who want to develop stuff which interact with Dogecoin Core, directly or more indirectly, may use it.

This balance between centralization and decentralization is pretty difficult. In some ways, I agree. I wouldn't want to see this repo as a *"one-stop shop for everything related to Dogecoin"*. I'm interested in open source and would like to see multiple project created around it. It's more about empowering the ecosystem. Coinbase don't really need our help for sure, it may be more for small projects to start with Dogecoin. It may also be a way to suggest other smaller and less known but still recommendable alternative, supporting them at the same time, helping in some decentralization ?

Idk what would be something truly decentralized. There isn't always some form of centralization ? If there are major features to implement in the future, with scalability for example, development & discussions would occur somewhere, centralizing things.

For Dogecoin Core, from my pov, a part of the decentralization would be to bring more people here to work on the core and taking part into discussions to reduce individuals power, by facilitating inclusion and opening discussions as much as possible.
But it will inevitably introduce some form of centralization by using Github.

*What is a decentralized decision ? You have 4 hours.*

**Comment by patricklodder** [comment]
> What is a decentralized decision for a "public domain phenomenon" ? 

Just do it under your own control: you make the decision, my opinion doesn't matter. Nothing prevents you from making a dogecoin-docs repository and do the work. Instead, the fact that you're here doing a proposal for something that doesn't necessarily need to be proposed here and we're having a discussion about it, already indicates that there's friction. There doesn't have to be any. 

Many centralized opinions and solutions that are interoperable make the whole decentralized. Having multiple channels keeps things permissionless. You don't have to argue with anyone to reach your goal, so why are you?

**Comment by AbcSxyZ** [comment]
For so many decisions, there are "frictions", this repo is having some necessary. Even technically speaking, in [1.21 consensus plan](https://github.com/dogecoin/dogecoin/discussions/2264) about Segwit there is debate, also to switch to proof of whatever.
Things can be discussed, it's what I wanted to do :)

It's possible to do a dogecoin-docs repo, but I guess it make sense to have the documentation about how to install & use the software with it ?
And because it's open source, explain how people can contribute to Dogecoin Core.

From my pov, I would like to see more people coming here to work on the code. Isn't that something to do?

**Comment by patricklodder** [comment]
> Even technically speaking, in 1.21 consensus plan about Segwit there is debate, also to switch to proof of whatever...

So you're saying that me summarizing and taking public a private discussion between maintainers of this repository on a feature that has historically been implemented through this software (consensus rules) with the goal to make the discussion broader and get feedback, is comparable to C&P-ing a tweet from a C-level exec of an entertainment conglomerate that claims to not have done any research before doing a poll (have to admit, that **_is_** funny), then triggering you to propose documenting this even though that has never been the function of this repository?!? I disagree with that being comparable, sorry.

> I guess it make sense to have the documentation about how to install & use the software with it ?

Yes. That's why we're working on that, and why those PRs get merged. But order processing is not something this software does. Just like having a bank account doesn't out-of-the-box give you that. Doing a monolithic approach to Dogecoin, centralizing everything around Dogecoin Core, is imho dangerous because it will make it unmaintainable over time. Also, just because some people elevate the importance of Dogecoin Core to make themselves look more important, doesn't mean this software or its repository should solve every problem. That will only create strenuous situations for a dev team that is already for 3/4th claiming to be burned out from all the hard work done here.

> From my pov, I would like to see more people coming here, on this GitHub repository, to work on the code.

If someone is implementing an order-to-cash software using Dogecoin Core and there is something missing in Dogecoin Core, sure. I'd love the contribution too. That doesn't mean we need to tell them how to do their job on other software they write, though. 


Bottom line... if you want to make it easier for developers and merchants to accept Dogecoin, then go ahead. Can even do the Pull Request here when you're done. I stay of the opinion that we should not try and centrally manage the ecosystem here, for reasons stated above.

**Comment by AbcSxyZ** [comment]
Not now, not here, not that way. The idea remain but I close this issue.
